jwst/dark_current/dark_sub.py

```diff
@@ -178,23 +275,38 @@ def subtract_dark(input, dark):
 
     """
 
+    dark_ints = dark.data.shape[0] -1
+
     log.debug("subtract_dark: nints=%d, ngroups=%d, size=%d,%d",
               input.meta.exposure.nints, input.meta.exposure.ngroups,
               input.data.shape[-1], input.data.shape[-2])
 
+
     # Create output as a copy of the input science data model
     output = input.copy()
 
+    
     # combine the science and dark DQ arrays
-    output.pixeldq = np.bitwise_or(input.pixeldq, dark.dq)
+    detector = input.meta.instrument.detector
+    if(detector[:3] == 'MIR'):
+           output.pixeldq = np.bitwise_or(input.pixeldq, np.bitwise_or(dark.dq[0,0,:,:], dark.dq[1,0,:,:]))
+    else:
+           output.pixeldq = np.bitwise_or(input.pixeldq, dark.dq)
 
 
     # loop over all integrations and groups in input science data
     for i in range(input.data.shape[0]):
-        for j in range(input.data.shape[1]):
+        if(detector[:3] =='MIR'):
+
+            if(i <= dark_ints): dark_int = dark.data[i]
```

Comment:
Ah right. I was missing the fact that it was selecting dark.data[i], which will obviously increment along with each execution of the loop. So the idea is to have it use the first nints of the dark data one-to-one with the first nints of the science data, and then after that just keep using the last int of the dark data for the remaining science ints.